runtime/lua/vim/lsp/handlers.lua

```diff
@@ -24,6 +24,48 @@ M['workspace/executeCommand'] = function(err, _)
   end
 end
 
+-- @msg of type ProgressParams
+-- Basically a token of type number/string
+local function progress_callback(_, _, params, client_id)
+  local client = vim.lsp.get_client_by_id(client_id)
+  if not client then
+    err_message("LSP[", client_id, "] client has shut down after sending the message")
+  end
+  local val = params.value    -- unspecified yet
+  local token = params.token  -- string or number
+
+
+  if val.kind then
+    if val.kind == 'begin' then
+      client.messages.progress[token] = {
+        title = val.title,
+        message = val.message,
+        percentage = val.percentage,
+        -- spinner = 1
```

Comment:
leftover from lsp-status

Reply:
It would be nice to have some sort of message count in order to implement a spinner, the percentage could be used but wouldn't be as smooth.

Reply:
removed. did you have a chance to test this PR ?

Reply:
I test with this crude https://github.com/teto/home/blob/master/config/nvim/lua/statusline.lua . 
The spinner could be added later on.

Reply:
Tried it out with: https://github.com/sharksforarms/lsp-status.nvim/compare/sharks/status-spinner...sharksforarms:sharks/status-spinnerz